generator/generate.go

```diff
@@ -388,13 +409,18 @@ func GenerateReports(res *SuiteResult, reportsDir, themePath string, searchIndex
 				slice++
 				relPath, _ := filepath.Rel(projectRoot, r.FileName)
 				env.CreateDirectory(filepath.Join(reportsDir, filepath.Dir(relPath)))
-				sf, err := os.Create(filepath.Join(reportsDir, toHTMLFileName(r.FileName, projectRoot)))
+				htmlFileName := filepath.Join(reportsDir, toHTMLFileName(r.FileName, projectRoot))
+				sf, err := os.Create(htmlFileName)
 				if err != nil {
 					return err
 				}
 				wg.Add(1)
 				propogateBasePath(r)
-				go generateSpecPage(res, r, sf, &wg)
+				go func(suiteRes *SuiteResult, specRes *spec, wc io.WriteCloser, htmlFileName string, wg *sync.WaitGroup) {
+					generateSpecPage(suiteRes, specRes, wc, wg)
+					minifyHTML(htmlFileName)
```

Comment:
instead of minifying each file individually, I suggest we generate the report as it is, and do a minification post this. Worth logging this step in `DEBUG` log level too. 

This will help handle scenario where one or more of file minification woul fail.

This shall also help keep the toggle at one place.